### PR TITLE
Add non-null constraints to foreign keys

### PR DESCRIPTION
**Pull request by YaleChen299**

### CODE REVIEW

**Review by RichDom2185 — COMMENTED** [review]
One of the migration files was in the wrong format (and thus skipped) but somehow the test cases still passed... Let's look into it before merging.

File: model/stories_test.go [hunk 1]
```diff
@@ -104,6 +104,17 @@ func TestCreateStory(t *testing.T) {
...
+	t.Run("cannot create without author", func(t *testing.T) {
+		db, cleanUp := testutils.SetupDBConnection(t, dbConfig, migrationPath)
+		defer cleanUp(t)
+
+		story := Story{
+			Content: "# Hi\n\nThis is a test story.",
+		}
+		err := CreateStory(db, &story)
+		assert.Error(t, err, "Expected error when creating story without Author ID")
+	})
```
Comment (RichDom2185) [inline model/stories_test.go:117]: I think we can do better and check for the error specifically being an FK violation or a null constraint violation.

File: model/usergroups_test.go [hunk 2]
```diff
@@ -81,6 +81,37 @@ func TestCreateUserGroup(t *testing.T) {
...
+		err := CreateUserGroup(db, &userGroup)
+		assert.Error(t, err, "Expected error when creating usergroup without group")
+
+		userGroup2 := UserGroup{
+			Group: group,
+			Role:  groupenums.RoleStandard,
+		}
+		err = CreateUserGroup(db, &userGroup2)
+		assert.Error(t, err, "Expected error when creating usergroup without group")
+	})
```
Comment (RichDom2185) [inline model/usergroups_test.go:114]: Ditto on checking the error type.

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
> One of the migration files was in the wrong format (and thus skipped) but somehow the test cases still passed... Let's look into it before merging.

I looked into it... the tests still passed because we enforce a non-null constraint in the model layer (because it's typed `uint` not `*uint`). So regardless of whether or not the DB has a non-null constraint, inserting empty will fail due to 0 being an FK violation.

Even with the migration file fixed, the insert fails not because of the null constraint violation, but instead because of the FK violation (inserting 0).

I think we should use `db.Exec` or `db.Raw` instead of relying on the model to test. That way, we can test for both behaviors: null  constraint violation as well as FK violation.

Thoughts? @YaleChen299

**Comment by YaleChen299** [comment]
> > One of the migration files was in the wrong format (and thus skipped) but somehow the test cases still passed... Let's look into it before merging.
> 
> I looked into it... the tests still passed because we enforce a non-null constraint in the model layer (because it's typed `uint` not `*uint`). So regardless of whether or not the DB has a non-null constraint, inserting empty will fail due to 0 being an FK violation.
> 
> Even with the migration file fixed, the insert fails not because of the null constraint violation, but instead because of the FK violation (inserting 0).
> 
> I think we should use `db.Exec` or `db.Raw` instead of relying on the model to test. That way, we can test for both behaviors: null constraint violation as well as FK violation.
> 
> Thoughts? @YaleChen299

Yup I agree. I will do it. I was just a bit lazy hahaha.

**Pull request merged**